docs/extensions/payment-processors/create.md

```diff
@@ -0,0 +1,734 @@
+   */
+  static protected $_mode = null;
+
+  /**
+   * Constructor
+   *
+   * @param string $mode the mode of operation: live or test
+   *
+   * @return void
+   */
+  function __construct( $mode, &$paymentProcessor ) {
+    $this->_mode             = $mode;
+    $this->_paymentProcessor = $paymentProcessor;
+    $this->_processorName    = ts('UC Merced Payment Collection');
+  }
+
+  /**
+   * singleton function used to manage this object
+   *
+   * @param string $mode the mode of operation: live or test
+   *
+   * @return object
+   * @static
+   *
+   */
+  static function &singleton( $mode, &$paymentProcessor ) {
+      $processorName = $paymentProcessor['name'];
+      if (self::$_singleton[$processorName] === null ) {
+          self::$_singleton[$processorName] = new edu_ucmerced_payment_ucmpaymentcollection( $mode, $paymentProcessor );
+      }
+      return self::$_singleton[$processorName];
+  }
+
+  /**
+   * This function checks to see if we have the right config values
+   *
+   * @return string the error message if any
+   * @public
+   */
+  function checkConfig( ) {
+    $config = CRM_Core_Config::singleton();
+
+    $error = array();
+
+    if (empty($this->_paymentProcessor['user_name'])) {
+      $error[] = ts('The "Bill To ID" is not set in the Administer CiviCRM Payment Processor.');
+    }
+
+    if (!empty($error)) {
+      return implode('<p>', $error);
+    }
+    else {
+      return NULL;
+    }
+  }
+
+  function doDirectPayment(&$params) {
+    CRM_Core_Error::fatal(ts('This function is not implemented'));
+  }
+
+  /**
+   * Sets appropriate parameters for checking out to UCM Payment Collection
+   *
+   * @param array $params  name value pair of contribution datat
+   *
+   * @return void
+   * @access public
+   *
+   */
+  function doTransferCheckout( &$params, $component ) {
+
+  }
+}
+```
+
+
+There are 4 areas that need changing for your specific processor.
+
+The class name needs to match to your chosen class name (directory of
+the extension)
+
+```php
+class edu_ucmerced_payment_ucmpaymentcollection extends CRM_Core_Payment {
+```
+
+The processor name needs to match the name of your processor.
+
+```php
+function __construct( $mode, &$paymentProcessor ) {
+  $this->_mode             = $mode;
+  $this->_paymentProcessor = $paymentProcessor;
+  $this->_processorName    = ts('UC Merced Payment Collection');
+}
+```
+
+This method should call your class name
+
+```php
+static function &singleton( $mode, &$paymentProcessor ) {
+    $processorName = $paymentProcessor['name'];
+    if (self::$_singleton[$processorName] === null ) {
+        self::$_singleton[$processorName] = new edu_ucmerced_payment_ucmpaymentcollection( $mode, $paymentProcessor );
+    }
+    return self::$_singleton[$processorName];
+}
+```
+
+You need to process the data given to you in the doTransferCheckout()
+method. Here is an example of how it's done in this processor
+
+
+**UCMPaymentCollection.php-doTransferCheckout**
```

Comment:
We need to say something about the differences between doTransferCheckout() and doDirectPayment().  doTransferCheckout() is not implemented in quite a few processors and vice-versa.  I think it depends on whether payment is taken indirectly, eg. sagepay, paypal or "directly", eg smartdebit, stripe